### PR TITLE
fix(i18n): update Chinese translation of functional programming

### PR DESCRIPTION
**Pull request by ZhichengChen**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

### CODE REVIEW

File: curriculum/challenges/chinese/02-javascript-algorithms-and-data-structures/functional-programming/understand-functional-programming-terminology.chinese.md [hunk 1]
```diff
@@ -2,29 +2,40 @@
...
 ---
 
 ## Description
-<section id="description">联邦通信委员会团队有一种情绪波动，现在想要两种类型的茶：绿茶和红茶。一般事实：客户情绪波动很常见。有了这些信息，我们需要重新审视上次挑战中的<code>getTea</code>功能，以处理各种茶叶请求。我们可以修改<code>getTea</code>来接受一个函数作为参数，以便能够改变它准备的茶的类型。这使得<code>getTea</code>更加灵活，并且在客户端请求发生变化时为程序员提供更多控制。但首先，让我们介绍一些函数术语： <code>Callbacks</code>函数是滑动或传递给另一个函数来决定函数调用的函数。您可能已经看到它们传递给其他方法，例如在<code>filter</code> ，回调函数告诉JavaScript如何过滤数组的标准。可以分配给变量，传递到另一个函数或从其他函数返回的函数就像任何其他正常值一样，称为<code>first class</code>函数。在JavaScript中，所有函数都是<code>first class</code>函数。将函数作为参数或将函数作为返回值返回的函数称为<code>higher order</code>函数。当函数传递给另一个函数或从另一个函数返回时，那些传入或返回的函数可以称为<code>lambda</code> 。 </section>
+<section id='description'>
+FCC 团队需求有变更，现在想要两种茶：绿茶（green tea）和红茶（black tea）。事实证明，用户需求变更是很常见的。
+基于以上信息，我们需要重构上一节挑战中的<code>getTea</code>函数来处理多种茶的请求。我们可以修改<code>getTea</code>接受一个函数作为参数，使它能够修改茶的类型。这让<code>getTea</code>更灵活，也使需求变更时为程序员提供更多控制权。
+首先，我们将介绍一些术语：
+<code>Callbacks</code>是被传递到另一个函数中调用的函数。你应该已经在其他函数中看过这个写法，例如在<code>filter</code>中，回调函数告诉 JavaScript 以什么规则过滤数组。
+函数就像其他正常值一样，可以赋值给变量、传递给另一个函数，或从其它函数返回，这种函数叫做<code>first class</code>函数。在 JavaScript 中，所有函数都是<code>first class</code>函数。
```
Comment (miyaliu666) [inline curriculum/challenges/chinese/02-javascript-algorithms-and-data-structures/functional-programming/understand-functional-programming-terminology.chinese.md]: 两处：<code>first class</code> => <code>头等</code>

File: curriculum/challenges/chinese/02-javascript-algorithms-and-data-structures/functional-programming/understand-functional-programming-terminology.chinese.md [hunk 2]
```diff
@@ -2,29 +2,40 @@
...
+将函数为参数或返回值的函数叫做<code>higher order</code>函数。
```
Comment (miyaliu666) [inline curriculum/challenges/chinese/02-javascript-algorithms-and-data-structures/functional-programming/understand-functional-programming-terminology.chinese.md]: <code>higher order</code> => <code>高阶</code>

**Review by miyaliu666 — APPROVED** [review]

**Comment by RandellDawson** [comment]
@ZhichengChen Please address the numerous merge conflicts present on this PR.

Thank you.

**Comment by ZhichengChen** [comment]
@RandellDawson I have addressed the merge conflicts.

**Review by raman0716 — APPROVED** [review]

**Pull request merged**